overlord/snapstate/handlers.go

```diff
@@ -324,19 +324,28 @@ func (m *SnapManager) installOneBaseOrRequired(t *state.Task, snapName string, c
 		return nil, err
 	}
 
-	inProgress := func(snapName string) (bool, error) {
+	shouldAttemptInstall := func(snapName string) (bool, error) {
 		linkTask, err := findLinkSnapTaskForSnap(st, snapName)
 		if err != nil {
 			return false, err
 		}
 
 		if linkTask == nil {
 			// snap is not being installed
+			return true, nil
+		}
+
+		// if we are remodeling, then we should return early due to the way that
+		// tasks are ordered by the remodeling code. specifically, all snap
+		// downloads during a remodel happen prior to snap installation. thus,
+		// we cannot wait for snaps to be installed here. see remodelTasks for
+		// more information on how the tasks are ordered.
```

Comment:
wont' this be a problem also for updatePrereqIfOutdated code? doesn't it mean we should simply check this before all other checks and return early? instead of doing the check here?

Reply:
I spoke to Miguel about this earlier, because I was also confused about the ordering of the original code:
> Those are mutually exclusive, unless I'm misunderstanding the question. The first if runs when the prerequisite is a default provider and the rest runs if it's another type of dependency (base, snapd). So that checks doesn't run after updating. Also the updatePrereqIfOutdated has code to check if its already being updated so it would catch a conflict

Reply:
Just to clarify, that comment was about the `inProgress` check vs the `updatePrereqIfOutdated` call which was the code snippet in the link. It wasn't about the remodel check

Reply:
Hmm I see I was conflating them.